optuna/storages/journal/_file.py

```diff
@@ -128,26 +129,52 @@ class JournalFileSymlinkLock(BaseJournalFileLock):
     Args:
         filepath:
             The path of the file whose race condition must be protected.
+        grace_period:
+            Grace period before an existing lock is forcibly released.
     """
 
-    def __init__(self, filepath: str) -> None:
+    def __init__(self, filepath: str, grace_period: int = 30) -> None:
```

Comment:
What about allowing `grace_period = None` (i.e., unlimited)? Forcing the release of long-held locks seems like a good practice in many cases, but I'm not sure it's always OK.